permhash/helpers.py

```diff
@@ -205,11 +241,40 @@ def create_crx_manifest_permlist(path):
     if check_type(path, CRX_MANIFEST_MIMETYPES):
         with open(path, encoding="utf-8") as manifest:
             try:
-                manifest_json = json.load(manifest)
```

Comment:
Dealing with the text encodings is getting pretty complicated and it looks like this code is largely the same as what you see in `create_crx_permlist`. Updating it in the future to deal with new possible encoding issues could be problematic.

My recommendation would be to have both `create_crx_manifest_permlist` and `create_crx_permlist` to get the manifest file into a bytestring (don't mess with encodings) and then  call a function called `process_manifest_bytes` which takes the bytestring and tries to decode and do all the json processing. Then both `create_crx_manifest_permlist` and `create_crx_permlist`  can call parse_crx_manifest as normal

Reply:
Agree. I was writing each part and they seemed to be coming together in a _different way_ but are handling the same data. Function created. Will be posted soon.